rc.js

```diff
@@ -29,6 +29,8 @@ if (env.npm_config_argv) {
 // Get the configuration
 module.exports = function (pkg) {
   var pkgConf = pkg.config || {}
+  var sourceBuild = env.npm_config_build_from_source
+  var buildFromSource = sourceBuild === pkg.name || (sourceBuild === true || sourceBuild === 'true')
```

Comment:
```suggestion
  var buildFromSource = sourceBuild === pkg.name || sourceBuild === 'true'
```

Because as it stands, the value is always a string.